test/Tests/src/Semantic/Java_Interop_Spec.enso

```diff
@@ -22,6 +22,10 @@ spec =
             list = ArrayList.new
             list.add 432
             list.get 0 . should_equal 432
+        Test.specify "should report missing method error on Java Arrays" <|
+            list = ArrayList.new
+            list.add 432
```

Comment:
Shall this operation be allowed at all? Shouldn't we rather require `Polyglot.invokeMember(list, 'add', [ 432 ])`? I am less and less convinced that supporting these polyglot operations on array makes sense. CCing @kustosz, @jdunkerley, @radeusgd

Reply:
I think this is a bit controversial.

1. On one hand, if I'm writing code using some Java library that returns some array-like object like `ArrayList`, I probably expect to be able to use its `ArrayList` API, as I would with any other polyglot Java object. Making it magical and require `invokeMember` is a bit inconsistent and I'd be surprised as a user.
2. On the other hand, if we see it's an array and sprinkle our special methods on it (`length`, `sort`, I guess in the future `map`, `filter` etc.) then it can become confusing if my original object also had a method with such a name - which one is selected (I assume the Enso ones take precedence, but that is not something immediately clear just from intuition but rather something that should be documented somewhere).

So in either way there is going to be some confusion - I think the way to go is to prepare a piece of documentation - _Enso Polyglot Programming Handbook_ which would definitely be needed once we expect more advanced users writing libraries, but could be immediately useful when onboarding new team members. There we could explain such nuances and then regardless of the decision it should be a bit easier to know what to do.

As for the `invokeMember` or not - I personally prefer to be able to just call these methods like on any other objects, knowing there are some Enso-specific overrides (they exist for _all_ objects anyway - like `Any.==` or `Any.is_nothing` etc. etc.) rather than having to use special `Polyglot.invokeMember`. But that is just my opinion, and I think both solutions are ok in general.